apps/studio/prisma/schema.prisma

```diff
@@ -198,6 +201,19 @@ model SiteMember {
   @@id([siteId, userId])
 }
 
+model ResourcePermission {
```

Comment:
Since we might need to extend this to future extended perms, what might be confusing is:

is null on resourceId = no resource that the user has access to 
or
is it to all resources (which is the case here) -> implicit allowance on a site-level

would it be better to introduce a enum that makes this entry clear

permsType - site vs resource

Under Site model, maybe we can add the perms type configured for the site - site wide vs resource level

Reply:
i think `resource === null` to refer to the root resource is pretty much locked in and we've quite a few places that use this convention too. i think the confusion part isn't as bad because if the user has no access, they won't even have the permission in the first place.

additionally, adding the `enum` complicates quite a few things
1. flipping from `site` -> `root` level permissions (another bit to flip)
2. checks next time against whether a user has access to x resource cannot be done with a single query to `ResourcePermission` but needs an additional query (and logic) against `Site` 

the permission checking stuff is hidden behind a library (casl) w. a conditional, so most of the time devs won't be interacting with this

Reply:
I was referring to resourceId === null being interpreted as site-wide perms. Is an implicit convention

Regarding complications:
1. Actually this bit will at most flip once (manually) when a site moves to extended perms, so not really complicated
2. This I agree as it adds 1 more DB touchpoint from Site table which will increases our authz latency

I think it's a fair point this interpretation is abstracted away by our library, hence am okay with this approach esp for point (2) you raised